tests/semantic_analyser.cpp

```diff
@@ -533,6 +533,14 @@ kprobe:f { @x = pid < 10000 ? "lo" : 2 }
 
 TEST(semantic_analyser, mismatched_call_types)
 {
+  // This assignment requires a cast, hint the requirement in the error message
+  test_error("kprobe:f { @x = 1; @y = count(); @x = @y; }", R"(
+stdin:1:34-36: ERROR: Type mismatch for @x: trying to assign value of type 'count_t' when map already contains a value of type 'int64', cast the 'count_t' value to 'int64'
```

Comment:
Bit of a long sentence. How about this instead?

```suggestion
stdin:1:34-36: ERROR: Type mismatch for @x: trying to assign value of type 'count_t' when map already contains a value of type 'int64'. Maybe cast the 'count_t' value to 'int64'?
```